Marlin/src/lcd/extui/lib/anycubic_chiron/chiron_tft.cpp

```diff
@@ -180,9 +181,10 @@ namespace Anycubic {
     #if ACDEBUG(AC_MARLIN)
       SERIAL_ECHOLNPAIR("ConfirmationRequest() ", msg, " printer_state:", printer_state);
     #endif
+    size_t msgH = HASH(msg);
     switch (printer_state) {
       case AC_printer_pausing: {
-        if (strcmp_P(msg, MARLIN_msg_print_paused) == 0 || strcmp_P(msg, MARLIN_msg_nozzle_parked) == 0) {
+        if (msgH == MARLIN_msg_print_paused || msgH == MARLIN_msg_nozzle_parked) {
```

Comment:
Actually, for this method, either it should be taking a flag / enum instead of comparing a string, or it should be simply comparing the passed string pointer to the address of the expected strings. We should not be too respectful of code from 3rd party vendors, which is often thrown together without any thoughts of optimization or future maintainability.

Reply:
I haven't found where are the string coming from (there are defined once but not used anywhere else). 
So I've deduced they come from an external peripheral (the chiron TFT system ?).

So I've modified the define to store the hash of the string instead of the strings. They are quite big so this gives quite a saving here.

If we can modify the source code from the peripheral, then yes an enum would be better. 

In fact, the `switch { switch {` code could probably be reversed (only testing the message's hash here and skip the printer state's switch all together), since I think they've written the code this way to avoid doing too much strcmp here.

This would even provide more code saving.